File: javascript/hamming/hamming.js
export const compute = (dna1, dna2) => {
  if (dna1.length !== dna2.length) 
    throw new Error('strands must be of equal length')

  //let distance = 0;
  //[...dna1].forEach((letter, index) => { distance += (letter === dna2[index] ?
  //                                                    0 : 1)});

  return [...dna1].reduce((distance, letter, index) => 
    distance + (letter === dna2[index] ? 0 : 1), 0);
};

File: python/hamming/hamming.py
def distance(strand_a, strand_b):
    la, lb = len(strand_a), len(strand_b)
    if la != lb:
        raise ValueError("Strands must be of equal length.")

    return sum([1 for i in range(la) if strand_a[i] != strand_b[i]])
